compiler/src/dmd/statementsem.d

```diff
@@ -1810,6 +1810,26 @@ Statement statementSemanticVisit(Statement s, Scope* sc)
                 return setError();
             }
         }
+        else if (ps.ident == Id.mangle)
+        {
+            auto es = ps._body ? ps._body.isExpStatement() : null;
+            auto de = es ? es.exp.isDeclarationExp() : null;
+            if (!de)
+            {
+                ps.error("`pragma(mangle)` must be attached to a declaration");
+                return setError();
+            }
+            const se = ps.args && (*ps.args).length == 1 ? semanticString(sc, (*ps.args)[0], "pragma mangle argument") : null;
+            if (!se)
+            {
+                ps.error("`pragma(mangle)` takes a single argument that must be a string literal");
+                return setError();
+            }
+            import dmd.root.rmem : xarraydup;
+            const name = (cast(const(char)[])se.peekData()).xarraydup;
```

Comment:
is the duplication really necessary? why not pass a slice of the original string?

Reply:
No idea, it is copied from dsymbolsem, see `visitPragmaDecl`

Reply:
I think this is a remnant from when the code was written in C++.

Reply:
Do I remove both or maybe you take note of this for a future trivial PR ?

Reply:
I think that for the purpose of this PR you can try remove it just on the new site and we can leave other uses for a future PR.